homeassistant/components/sfr_box/diagnostics.py

```diff
@@ -26,13 +26,17 @@ async def async_get_config_entry_diagnostics(
             "data": dict(entry.data),
         },
         "data": {
-            "dsl": async_redact_data(dataclasses.asdict(data.dsl.data), TO_REDACT),
+            "dsl": async_redact_data(
+                dataclasses.asdict(await data.system.box.dsl_get_info()), TO_REDACT
+            ),
             "ftth": async_redact_data(
                 dataclasses.asdict(await data.system.box.ftth_get_info()), TO_REDACT
             ),
             "system": async_redact_data(
-                dataclasses.asdict(data.system.data), TO_REDACT
+                dataclasses.asdict(await data.system.box.system_get_info()), TO_REDACT
+            ),
+            "wan": async_redact_data(
+                dataclasses.asdict(await data.system.box.wan_get_info()), TO_REDACT
```

Comment:
This has been adjusted to get the latest data, instead of the potentially stale data.